### PR TITLE
Tabular Loan UI added

### LINKED ISSUE #3490: Add Separate Tabs for Pending and Completed Loans in the Loan Section
### Please confirm the following:

- [X] I've checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate issues.
- [X] I've requested a single (only one) feature/change in this issue. It complies with the One Request Per GitHub Issue (ORPGI) rule.
- [X] My issue is well-defined and describes how it should be implemented from UI/UX perspective.

### What do you want to be added or improved?

It would be helpful to introduce two distinct tabs in the loan section for improved organization and clarity.
Pending Loans: A tab that lists loans that are still active or in progress.
Completed Loans: A tab that lists loans that have been fully paid off or closed.
This feature would help users easily differentiate between their current obligations and those that have been completed. By having these two tabs, it would make the loan section more user-friendly and improve overall navigation.

### Why do you need it?

Proposed Solution:
Add two separate tabs within the loan section:
Pending Loans: For loans that are ongoing or still require payments.
Completed Loans: For loans that have been fully paid off or closed.

This way, users can easily view and manage their ongoing loans without having to filter through completed ones.

Benefits:
Improved clarity: Users can quickly identify which loans are active.
Easier tracking: Allows users to focus on the loans that require attention.
Organized interface: Reduces clutter and improves usability.

### How do you imagine it?

_No response_

### PR DESCRIPTION
**Pull request by shamim-emon**
## Pull request (PR) checklist

Please check if your pull request fulfills the following requirements:

- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [x] I've read and understand the [Developer Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Guidelines.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [x] My PR includes only the necessary changes to fix the issue (i.e., no unnecessary files or lines of code are changed).
- [x] 🎬 I've attached a **screen recording** of using the new code to the next paragraph (if applicable).

## Screen recording of your changes (if applicable):

https://github.com/user-attachments/assets/d960d8be-e40d-4fa5-853b-77abda8cc5f9

## What's changed?

Describe with a few bullets **what's new:**
- I've added optional tabular loan screen, where user can see pending & completed loans in separate tabs.

## Risk factors

**What may go wrong if we merge your PR?**
- N/A
**In what cases won't your code work?**
- N/A

## Does this PR close any GitHub issues? (do not delete)

- Closes #3490 

## Troubleshooting GitHub Actions (CI) failures ❌
Pull request checks failing? Read our [CI Troubleshooting guide](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/CI-Troubleshooting.md).

### CODE REVIEW

File: screen/loans/src/main/java/com/ivy/loans/loan/LoanScreenState.kt [hunk 1]
```diff
@@ -9,12 +9,16 @@ import java.time.Instant
...
     val reorderModalVisible: Boolean,
     val totalOweAmount: String,
     val totalOwedAmount: String,
     val paidOffLoanVisibility: Boolean,
-    val dateTime: Instant
-)
+    val tabularLoanMode: Boolean,
+    val dateTime: Instant,
+    val selectedTab: LoanTab
+)
```
Comment (ILIYANGERMANOV) [inline screen/loans/src/main/java/com/ivy/loans/loan/LoanScreenState.kt:24]: Make the state a sealed interface with two cases - tabular and regulad

File: screen/loans/src/main/java/com/ivy/loans/loan/LoansScreen.kt [hunk 2]
```diff
@@ -100,46 +103,88 @@ private fun BoxWithConstraintsScope.UI(
...
+
+                NoLoansEmptyState(
+                    emptyStateTitle = stringResource(R.string.no_loans),
+                    emptyStateText = stringResource(R.string.no_loans_description)
                 )
+
+                Spacer(Modifier.weight(1f))
+            }
+        } else {
+            for (item in state.loans) {
```
Comment (ILIYANGERMANOV) [inline screen/loans/src/main/java/com/ivy/loans/loan/LoansScreen.kt:138]: Use LazyColumn instead of `for` because it's more efficient

File: screen/loans/src/main/java/com/ivy/loans/loan/LoansScreen.kt [hunk 3]
```diff
@@ -78,7 +77,7 @@ fun BoxWithConstraintsScope.LoansScreen(screen: LoansScreen) {
 @Composable
 private fun BoxWithConstraintsScope.UI(
     state: LoanScreenState,
-    onEventHandler: (LoanScreenEvent) -> Unit = {},
+    onEventHandler: (LoanScreenEvent) -> Unit = {}
```
Comment (ILIYANGERMANOV) [inline screen/loans/src/main/java/com/ivy/loans/loan/LoansScreen.kt:80]: Add screenshot tests for all cases of this UI (regular, tabular, empty state)

**Review by ILIYANGERMANOV — CHANGES_REQUESTED** [review]
Didn't review much in detail but overall LGTM! Left some comments

**Review by suyash01 — COMMENTED** [review]
LGTM

**Comment by ILIYANGERMANOV** [comment]
> LGTM except what Iliyan mentioned.

@suyash01 thank you for reviewing! I'm planning to delegate reviewers to contributors so our project is even more community-driven

File: screen/loans/src/main/java/com/ivy/loans/loan/LoansScreen.kt (hunk 3 above)
Comment (shamim-emon) [reply screen/loans/src/main/java/com/ivy/loans/loan/LoansScreen.kt:81]: Done

File: screen/loans/src/main/java/com/ivy/loans/loan/LoansScreen.kt (hunk 2 above)
Comment (shamim-emon) [reply screen/loans/src/main/java/com/ivy/loans/loan/LoansScreen.kt]: Done

File: screen/loans/src/main/java/com/ivy/loans/loan/LoanScreenState.kt (hunk 1 above)
Comment (shamim-emon) [reply screen/loans/src/main/java/com/ivy/loans/loan/LoanScreenState.kt:24]: Done

Comment (ILIYANGERMANOV) [reply screen/loans/src/main/java/com/ivy/loans/loan/LoanScreenState.kt:24]: @shamim-emon I meant:
```kotlin
sealed interface LoanScreenViewState {
  // common fields

  data class Regular(
    val loans: ImmutableList<Loan>,
    override val for all common
   ) : LoanScreenViewState
   
   data class Tabular(
     val completedLoans: ImmutableList<DisplayLoan>,
    val pendingLoans: ImmutableList<DisplayLoan>,
   ) : LoanScreenViewState
}
```

This data modeling fixes the issue where "loans" is useless for the tabular UI and completed + pending is useless for the regular mode. See the data modeling guidelines in Ivy - this eliminates impossible cases and better represent reality. 

However, we're good to merge it. Tbh, your UI is better than the original so we might just get rid of the setting and maks the tabular UI the new Loans tab UI. I'll merge the PR, if you have time feel free to make the tabular UI the only Loans UI and cleanup the regular UI code

**Review by ILIYANGERMANOV — APPROVED** [review]

File: screen/loans/src/main/java/com/ivy/loans/loan/LoanScreenState.kt (hunk 1 above)
Comment (shamim-emon) [reply screen/loans/src/main/java/com/ivy/loans/loan/LoanScreenState.kt:24]: @ILIYANGERMANOV I have taken up the task. Will get rid of the old loan UI & relevant code.

File: screen/loans/src/main/java/com/ivy/loans/loan/LoanViewModel.kt [hunk 4]
```diff
@@ -332,6 +375,14 @@ class LoanViewModel @Inject constructor(
         }
     }
 
+    private fun loadCompletedLoans() {
+        completedLoans = allLoans.filter { loan -> loan.percentPaid == 1.0 }.toImmutableList()
```
[the remark this answers is not in the document]
Comment (shamim-emon) [reply screen/loans/src/main/java/com/ivy/loans/loan/LoanViewModel.kt:379]: @nvllz sure
